Code/Framework/AzFramework/Platform/Linux/AzFramework/Input/Devices/Keyboard/InputDeviceKeyboard_xcb.cpp

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/std/reference_wrapper.h>
+#include <AzCore/std/sort.h>
+#include <AzCore/std/typetraits/integral_constant.h>
+#include <AzFramework/API/ApplicationAPI_Linux.h>
+#include <AzFramework/Input/Buses/Notifications/RawInputNotificationsBus_xcb.h>
+#include <AzFramework/Input/Devices/Keyboard/InputDeviceKeyboard.h>
+
+#define explicit ExplicitIsACXXKeyword
+#include <xcb/xkb.h>
+#undef explicit
+#include <xkbcommon/xkbcommon-keysyms.h>
+#include <xkbcommon/xkbcommon.h>
+#include <xkbcommon/xkbcommon-x11.h>
+
+namespace AzFramework
+{
+    class InputDeviceKeyboardXcb
+        : public InputDeviceKeyboard::Implementation
+        , public RawInputNotificationBusXcb::Handler
+    {
+    public:
+        AZ_CLASS_ALLOCATOR(InputDeviceKeyboardXcb, AZ::SystemAllocator, 0);
+
+        using InputDeviceKeyboard::Implementation::Implementation;
+        InputDeviceKeyboardXcb(InputDeviceKeyboard& inputDevice)
+            : InputDeviceKeyboard::Implementation(inputDevice)
+        {
+            RawInputNotificationBusXcb::Handler::BusConnect();
+
+            auto* connection = AzFramework::LinuxXcbConnectionManagerInterface::Get()->GetXcbConnection();
+
```

Comment:
Should add an AZ_Assert here for connection

Reply:
Fixed.